jenkins/dockerfiles/build-and-push.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Copyright (C) 2019 Linaro Limited
+# Author: Matt Hart <matthew.hart@linaro.org>
+#
+# This module is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 2.1 of the License, or (at your option)
+# any later version.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+# Rebuild all of the docker images for kernelci
+# You will need push privileges to kernelci namespace on dockerhub
+# -n = disable cache (for when rebuilding a moving target image e.g. clang-8)
+# -p = push to dockerhub
+# -b = also rebuild build-base
+# -d = also rebuild debos
+# -q = make the builds quiet
+
+set -e
+
+options='npbdq'
+while getopts $options option
+do
+  case $option in
+    n)  cache_args="--no-cache";;
+    p)  push=true;;
+    b)  base=true;;
+    d)  debos=true;;
+    q)  quiet="--quiet";;
+  esac
+done
+
+echo_build() {
+  echo "Building [$1]"
+}
+
+echo_push() {
+  echo "Pushing [$1]"
+}
+
+if [ "x${base}" == "xtrue" ]
+then
+  tag=kernelci/build-base
+  echo_build $tag
+  docker build ${quiet} ${cache_args} build-base -t $tag
+  if [ "x${push}" == "xtrue" ]
+  then
+    echo_push $tag
+    docker push $tag
+  fi
+fi
+
+for c in {gcc,clang}-*
+do
+  tag=kernelci/build-$c
```

Comment:
Can we have a way to use another base?  Something to replace `kernelci/`?  During development, I'm pushing my images to another repository in the Docker hub and I use a different `DOCKER_BASE` parameter in the Jenkins jobs to get these images.  I just have a local `kernelci/build-base` tag which I don't push, purely because all the other Dockerfiles derive from it.

Reply:
Good idea, I do the same. Will add

Reply:
Any update with this?  It would be nice to be able to use it during the next production update.